synapse/lib/ast.py

```diff
@@ -2656,7 +2659,37 @@ class EditPropSet(Edit):
                 runt.reqLayerAllowed(('prop:set', prop.full))
 
             try:
+
+                if isadd or issub:
+
+                    if not isinstance(prop.type, s_types.Array):
+                        mesg = f'Property set using ({oper}) is only valid on arrays.'
+                        raise s_exc.StormRuntimeError(mesg)
+
+                    arry = node.get(name)
+                    if arry is None:
+                        arry = ()
+
+                    if isadd:
+                        # this new valu will get normed by the array prop
+                        valu = tuple(arry) + (valu,)
```

Comment:
The arry value which comes back from the node may not be a tuple, but may be a list